openthread: add experimental TCP support (#9345)
nrfconnect/sdk-nrf: doc/nrf/ug_thread_supported_features.rst

Thread 1.3 Specification introduces DNS-SD Service Registration Protocol, which lets devices advertise the fact that they provide services while avoiding the use of multicast in the discovery.
|NCS| provides the required SRP client functionality.

Transport Control Protocol
==========================

> While TCP support has been present in |NCS| for a long time (:ref:`IP stack supported features <ip_stack_overview>`), the Thread 1.3 Specification defines a set of parameters and features that make TCP more efficient for the limited IEEE 802.15.4 networks.

Reviewer: ```suggestion
While |NCS| has had TCP support through Zephyr (:ref:`IP stack supported features <zephyr:ip_stack_overview>`), the Thread 1.3 Specification defines a set of parameters and features that make TCP more efficient for the limited IEEE 802.15.4 networks.
```

> An alteranitive TCP stack implementation incorporated from the OpenThread project can be enabled by users looking forward for Thread based TCP applications.

Reviewer: ```suggestion
An alternative TCP stack implementation incorporated from the OpenThread project can be enabled by users looking forward for Thread-based TCP applications.
```
I am not quite sure what is being said here. "To look forward to" is usually used to mean someone is waiting for something to happen, like "I am looking forward to our date".

Author: I rephrased it a bit.
